test/tcp_echo_client/tcp_echo_client.cpp

```diff
@@ -43,17 +227,38 @@ void app_start(int argc, char *argv[])
     } ip_addr = {0, 0, 0, 0};
     int port = 0;
 
-    printf("MBED: TCPCllient waiting for server IP and port...\r\n");
+    printf("MBED: SAL TCP Client waiting for server IP and port...\r\n");
     scanf("%d.%d.%d.%d:%d", &ip_addr.ip_1, &ip_addr.ip_2, &ip_addr.ip_3, &ip_addr.ip_4, &port);
```

Comment:
a struct sock_addr is not required and we can just deal with a dotted decimal string with port. Using setAddr is therefore not required. I have cleaned up the code to remove parsing struct s_ip_address